File: generate_keys.js
const jwt = require('jsonwebtoken');

const secret = 'super-secret-jwt-token-with-at-least-32-characters-length';

function generateKey(role) {
  const payload = {
    role: role,
    iss: 'supabase',
    iat: Math.floor(Date.now() / 1000),
    exp: Math.floor(Date.now() / 1000) + 3153600000 // 100 years
  };
  return jwt.sign(payload, secret, { algorithm: 'HS256' });
}

console.log(`JWT_SECRET=${secret}`);
console.log(`ANON_KEY=${generateKey('anon')}`);
console.log(`SERVICE_ROLE_KEY=${generateKey('service_role')}`);

File: generate_keys.py
import jwt
import time

secret = "super-secret-jwt-token-with-at-least-32-characters-length"

def generate_key(role):
    payload = {
        "role": role,
        "iss": "supabase",
        "iat": int(time.time()),
        "exp": int(time.time()) + 3153600000 # 100 years
    }
    return jwt.encode(payload, secret, algorithm="HS256")

print(f"JWT_SECRET={secret}")
print(f"ANON_KEY={generate_key('anon')}")
print(f"SERVICE_ROLE_KEY={generate_key('service_role')}")
